package/MDAnalysis/analysis/distances.py

```diff
@@ -31,13 +31,17 @@
 __all__ = ['distance_array', 'self_distance_array', 'contact_matrix', 'dist']
 
 import numpy as np
-from scipy import sparse
 
 from MDAnalysis.lib.distances import distance_array, self_distance_array
 from MDAnalysis.lib.c_distances import contact_matrix_no_pbc, contact_matrix_pbc
 
 import logging
 
+try:
+    from scipy import sparse      
+except ImportError:
+    sparse = None
+
```

Comment:
That's already good but can you please raise an `ImportError` where you would actually try to use `sparse`, with a message saying that this functionality requires the scipy package to be installed? You can catch the `AttributeError` and then re-raise or just test `sparse is None` (which might be less pythonic but perhaps clearer... not sure. None of this is very pretty but a compromise between usability and clarity of dependencies.)

I *think* `ImportError` is still the best exception to be used  but I could be swayed.